core/genesis.go

```diff
@@ -83,7 +84,12 @@ func (ga *GenesisAlloc) UnmarshalJSON(data []byte) error {
 // flush adds allocated genesis accounts into a fresh new statedb and
 // commit the state changes into the given database handler.
 func (ga *GenesisAlloc) flush(db ethdb.Database) (common.Hash, error) {
-	statedb, err := state.New(common.Hash{}, state.NewDatabase(db), nil)
+	triedb := state.NewDatabase(db)
+	snaps, err := snapshot.New(db, triedb.TrieDB(), 1, common.Hash{}, false, true, false)
```

Comment:
Personally I think it's a dangerous action. This function can be invoked by `ToBlock`, and it can be called everywhere. But state snapshot needs to be singleton, it's very risky to manipulate states in disk.

Reply:
I could check if the snapshot isn't already present, if you'd like?

I don't think that there is a problem with `ToBlock` being invoked from all over, because from what I see, it's only called if no state is present on mainnet. So I'm not sure I get what the danger would be here?